idx/memory/memory_test.go

```diff
@@ -534,6 +534,98 @@ func TestSingleNodeMetric(t *testing.T) {
 	ix.AddOrUpdate(data, 1)
 }
 
+func TestGetByTag(t *testing.T) {
+	tagSupport := TagSupport
+	defer func() { TagSupport = tagSupport }()
+	TagSupport = true
+
+	ix := New()
+	ix.Init()
+
+	mds := getMetricData(1, 2, 50, 10, "metric.public")
+	mds[10].Tags = []string{"key1=value1", "key2=value2"}
+	mds[21].Tags = []string{"key1=value1"}
+	mds[28].Tags = []string{"key1=value2", "key2=value2"}
+	mds[30].Tags = []string{"key1=value1", "key3=value3"}
+
+	for _, md := range mds {
+		ix.AddOrUpdate(md, 1)
+	}
+
+	expressions := [][]string{
+		{"key1=value1"},
+		{"key1=value2"},
+		{"key1=~value[0-9]"},
+		{"key1=~value[23]"},
+		{"key1=value1", "key2=value1"},
+		{"key1=value1", "key2=value2"},
+		{"key1=~value[12]", "key2=value2"},
+		{"key1=~value1", "key1=value2"},
+		{"key1=~value[0-9]", "key2=~", "key3!=value3"},
+		{"key2=", "key1=value1"},
+	}
+	expecting := []int{3, 1, 4, 1, 0, 1, 2, 0, 1, 2}
```

Comment:
let's make 1 slice of cases, where each case shows its expression and which id's are expected to match.
just checking the number of matches can hide bugs, is hard for readers, and having 2 structures is a bit confusing